crates/iceberg/src/error.rs

```diff
@@ -259,6 +264,25 @@ define_from_err!(
     "handling invalid utf-8 characters"
 );
 
+/// Helper macro to check arguments.
+///
+///
+/// Example:
+///
+/// Following example check `a > 0`, otherwise returns an error.
+/// ```ignore
+/// use iceberg::check;
+/// check!(a > 0, "{} is not positive.", a);
+/// ```
+#[macro_export]
+macro_rules! check {
```

Comment:
`check` is so general that could be confusion, how about take the idea `ensure` from anyhow: `ensure_data_valid`?